pkg/api/norman/customization/aks/handler.go

```diff
@@ -126,6 +140,52 @@ func (h *handler) ServeHTTP(writer http.ResponseWriter, req *http.Request) {
 	}
 }
 
+func (h *handler) checkCredentials(req *http.Request) (int, error) {
+	cred := &Capabilities{}
+	raw, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		return http.StatusBadRequest, fmt.Errorf("cannot read request body: %v", err)
+	}
+
+	if err = json.Unmarshal(raw, &cred); err != nil {
+		return http.StatusBadRequest, fmt.Errorf("cannot parse request body: %v", err)
+	}
+
+	if cred.SubscriptionID == "" {
+		return http.StatusBadRequest, fmt.Errorf("invalid subscriptionId")
+	}
+	if cred.ClientID == "" {
+		return http.StatusBadRequest, fmt.Errorf("invalid clientId")
+	}
+	if cred.ClientSecret == "" {
+		return http.StatusBadRequest, fmt.Errorf("invalid clientSecret")
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	if cred.TenantID == "" {
+		cred.TenantID, err = azureutil.FindTenantID(ctx, azure.PublicCloud, cred.SubscriptionID)
+		if err != nil {
+			return http.StatusBadRequest, err
+		}
+	}
+	if cred.BaseURL == "" {
+		cred.BaseURL = azure.PublicCloud.ResourceManagerEndpoint
+	}
+	if cred.AuthBaseURL == "" {
+		cred.AuthBaseURL = azure.PublicCloud.ActiveDirectoryEndpoint
+	}
+	client, err := NewSubscriptionServiceClient(cred)
+	if err != nil {
+		return http.StatusBadRequest, err
+	}
+	_, err = client.Get(ctx, cred.SubscriptionID)
+	if err != nil {
+		return http.StatusUnauthorized, err
+	}
```

Comment:
Would findTenantID succeed if this were to fail? Would it make sense to do this check first?

Reply:
No, this has to be the final step. This is just a dummy API call to test whether the credentials in the client are working. FindTenantID uses the subscription ID but no other credentials to look up the tenant ID, whereas this is using an authenticated client to make a generic GET request, and I happened to choose the subscriptions API to use as the test.